lib/index.d.ts

```diff
@@ -2076,8 +2076,8 @@ declare namespace Joi {
          * @param schema - the schema object.
          * @param message - optional message string prefix added in front of the error message. may also be an Error object.
          */
-        attempt(value: any, schema: Schema, options?: ValidationOptions): any;
-        attempt(value: any, schema: Schema, message: string | Error, options?: ValidationOptions): any;
+        attempt<T = any>(value: any, schema: Schema<T>, options?: ValidationOptions): T;
+        attempt<T = any>(value: any, schema: Schema<T>, message: string | Error, options?: ValidationOptions): T;
```

Comment:
This forces you to redefine types that are already defined on the schema itself. What do you think of this instead?
```suggestion
        attempt<TSchema extends Schema>(value: any, schema: TSchema, options?: ValidationOptions): TSchema extends Schema<infer Value> ? Value : never;
        attempt<TSchema extends Schema>(value: any, schema: Schema, message: string | Error, options?: ValidationOptions): TSchema extends Schema<infer Value> ? Value : never;
```